### DISCUSSION TITLE
I'd like to thank you.

### DISCUSSION THREAD

**Discussion by Snoowp** [algorithmicsuperintelligence/optillm#66] [General]
Using this repo to explore some AI Agentic Workflows I've made. And I'd like to thank you for creating this. Do you have any Discord handles? Also, how do we know what to use and when to use something? If we can map that I can create a self prompting optimization command chain that does this on the fly.

**Comment by codelion** [comment]
Hey you can join the patched discord - https://discord.gg/XDxA3mJyhE I usually hang around there. my handle is `codelion_`

For choosing which technique to use when, it is not clear it depends on the kind of task. What I have done is that I trained a bert-style classifier model to do the prediction based on the prompt and a effort value (0-1).  The code for that is available in the `router` plugin - https://github.com/codelion/optillm/blob/main/optillm/plugins/router_plugin.py You can use it as `router-gpt-4o-mini` and it will chosose the most appropriate approch based on the intial query. I am still testing it, so haven't had the time to write about it.
